contributed/ggStrongLensingDC2_SL_NanLi_HackurDC2.ipynb

```diff
@@ -0,0 +1,752 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "mags_data = ggsl_data.loadinMags()\n",
+    "smag_u = mags_data['mag_true_u_lsst']\n",
+    "smag_g = mags_data['mag_true_g_lsst']\n",
+    "smag_r = mags_data['mag_true_r_lsst']\n",
+    "smag_i = mags_data['mag_true_i_lsst']\n",
+    "smag_z = mags_data['mag_true_z_lsst']\n",
+    "del mags_data\n",
+    "\n",
+    "smag_u.astype('float32').tofile('smag_u.bin')\n",
+    "smag_g.astype('float32').tofile('smag_g.bin')\n",
+    "smag_r.astype('float32').tofile('smag_r.bin')\n",
+    "smag_i.astype('float32').tofile('smag_i.bin')\n",
+    "smag_z.astype('float32').tofile('smag_z.bin')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "morph_data = ggsl_data.loadinMorph()\n",
+    "sda    = morph_data['size_major_disk_true']\n",
+    "sdb    = morph_data['size_minor_disk_true']\n",
+    "sdn    = morph_data['sersic_disk']\n",
+    "sba    = morph_data['size_major_bulge_true']\n",
+    "sbb    = morph_data['size_minor_bulge_true']\n",
+    "sbn    = morph_data['sersic_bulge']\n",
+    "sfrac  = morph_data['bulge_to_total_ratio_i']\n",
+    "del morph_data\n",
+    "\n",
+    "sda.astype('float32').tofile('sda.bin')\n",
+    "sdb.astype('float32').tofile('sdb.bin')\n",
+    "sdn.astype('float32').tofile('sdn.bin')\n",
+    "sba.astype('float32').tofile('sba.bin')\n",
+    "sbb.astype('float32').tofile('sbb.bin')\n",
+    "sbn.astype('float32').tofile('sbn.bin')\n",
+    "sfrac.astype('float32').tofile('sfrac.bin')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "bcgs_data = ggsl_data.loadinBCGs()\n",
+    "ra_main  = bcgs_data['ra_true']\n",
+    "dec_main = bcgs_data['dec_true']\n",
+    "zl_main  = bcgs_data['redshift_true']\n",
+    "magg_main= bcgs_data['mag_true_g_lsst']\n",
+    "magr_main= bcgs_data['mag_true_r_lsst']\n",
+    "del bcgs_data\n",
+    "\n",
+    "ra_main.astype('float32').tofile('ra_main.bin') \n",
+    "dec_main.astype('float32').tofile('rdec_main.bin')  \n",
+    "zl_main.astype('float32').tofile('zl_main.bin')   \n",
+    "magg_main.astype('float32').tofile('magg_main.bin') \n",
+    "magr_main.astype('float32').tofile('magr_main.bin') \n",
+    "\n",
+    "z_ref = 2.0\n",
+    "vd_main = mags_to_vd(magg_main, magr_main, zl_main)\n",
+    "re_main = re_sv(vd_main,zl_main,z_ref)\n",
+    "\n",
+    "vd_main.astype('float32').tofile('vd_main.bin') \n",
+    "re_main.astype('float32').tofile('re_main.bin') "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "bsz_arc = 20.0\n",
+    "bsz_deg = bsz_arc/3600.0\n",
+    "fboost = 1.0\n",
+    "z_ref = 2.0\n",
+    "\n",
+    "n_runs = 1000000\n",
```

Comment:
Here it would be great to have more context/discussion/comment for the users (actually, more of a description about the various cells/the flow of the code would be useful more generally).

Can you describe what the functionality of each task is (eg. selecting galaxies for source and lens, computing the distortion to the image etc.)?

Reply:
I cleaned up the notebook and added more context, for example

* What the parameters stand for. 
* The functionality of each block of the codes. 

Hopefully, it will be helpful. Please do not hesitate to leave me a message, if you have any questions or suggestions. Thanks a lot for your time and efforts.